crates/thread_aware_cache/src/shard.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Cache shard implementation.
+//!
+//! Each shard contains a Swiss Table (`hashbrown::HashMap`) for storage and SIEVE metadata
+//! for eviction. Shards are cache-line aligned to prevent false sharing.
+
+use std::hash::{BuildHasher, Hash};
+
+use hashbrown::DefaultHashBuilder;
+use hashbrown::HashMap;
+use parking_lot::RwLock;
+
+use crate::sieve::{NodeIndex, SieveList};
+
+/// Cache line size for alignment to prevent false sharing.
+const CACHE_LINE_SIZE: usize = 64;
```

Comment:
Nobody is making a 128 byte cache line? There's no system-level constant for this we could leverage?